nix-ts-mode.el

```diff
@@ -151,18 +157,28 @@
    :language 'nix
    :feature 'function
    `((function_expression
-      ":" @font-lock-misc-punctuation-face)))
+      ":" @font-lock-misc-punctuation-face))
 
+   :language 'nix
+   :feature 'error
+   :override t
+   '((ERROR) @font-lock-warning-face))
```

Comment:
I'm also not convinced this is necessary. `ERROR` is the placeholder node for  when the grammar can't correctly parse a portion of the code into the tree, and personally, I take the lack of syntax highlighting in my code as a sign I've made a syntactic error.

On the other hand, I'm not sure if this is common with other `-ts-modes`. Any thoughts, @purcell?

Reply:
Actually this turns out to be used in _some_ built-in `-ts` modes like `ruby-ts-mode`, `rust-ts-mode` and `toml-ts-mode`, but _not_ `python-ts-mode` or `java-ts-mode`. So I'd conclude it's somewhat a matter of taste. I say let's add it — it makes parsing issues a bit more explicit.

Reply:
Interesting - no harm in leaving it in then!

Reply:
I think it's okay to add it. Since it's in the fourth decoration level, it's not enabled by default.